WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.kt

```diff
@@ -18,19 +22,55 @@ class ReaderTagHeaderView @JvmOverloads constructor(
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : RelativeLayout(context, attrs, defStyleAttr) {
-    private val binding = ReaderTagHeaderViewBinding.inflate(LayoutInflater.from(context), this, true)
+    private var binding: ReaderTagBinding
 
     @Inject
     lateinit var uiHelpers: UiHelpers
 
+    @Inject
+    lateinit var readerImprovementsFeatureConfig: ReaderImprovementsFeatureConfig
+
     private var onFollowBtnClicked: (() -> Unit)? = null
 
     init {
         (context.applicationContext as WordPress).component().inject(this)
+        binding = if (readerImprovementsFeatureConfig.isEnabled()) {
+            val readerTagHeaderViewNewBinding =
+                ReaderTagHeaderViewNewBinding.inflate(LayoutInflater.from(context), this, true)
+            ReaderTagBinding.ImprovementsEnabled(
+                textTag = readerTagHeaderViewNewBinding.textTag,
+                followButton = readerTagHeaderViewNewBinding.followContainer.followButton,
+                textTagFollowCount = readerTagHeaderViewNewBinding.followContainer.textBlogFollowCount,
+            )
+        } else {
+            val readerTagHeaderViewBinding =
+                ReaderTagHeaderViewBinding.inflate(LayoutInflater.from(context), this, true)
+            ReaderTagBinding.ImprovementsDisabled(
+                textTag = readerTagHeaderViewBinding.textTag,
+                followButton = readerTagHeaderViewBinding.followButton,
+            )
+        }
         binding.followButton.setOnClickListener { onFollowBtnClicked?.invoke() }
     }
 
+    abstract class ReaderTagBinding {
+        abstract val textTag: MaterialTextView
+        abstract val followButton: ReaderFollowButton
+
+        data class ImprovementsDisabled(
+            override val textTag: MaterialTextView,
+            override val followButton: ReaderFollowButton
+        ) : ReaderTagBinding()
+
+        data class ImprovementsEnabled(
+            override val textTag: MaterialTextView,
+            override val followButton: ReaderFollowButton,
+            val textTagFollowCount: MaterialTextView,
+        ) : ReaderTagBinding()
+    }
+
     fun updateUi(uiState: ReaderTagHeaderUiState) = with(binding) {
+        (binding as? ReaderTagBinding.ImprovementsEnabled)?.textTagFollowCount?.gone()
```

Comment:
Since there are plans to have a followers count label in the tag header in the future, I've decided to reuse the same layout as the site header and make the followers label gone until we implement it.